int32 num_inputs
int32 num_outputs
float32 reward
int32 last_action
float32[] input_state
float32[] resulting_state